src/DotNetWorker/Http/HttpResponseData.cs

```diff
@@ -1,25 +1,34 @@
-// Copyright (c) .NET Foundation. All rights reserved.
+﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the MIT License. See License.txt in the project root for license information.
 
-using System.Collections.Generic;
+using System.IO;
 using System.Net;
 
-namespace Microsoft.Azure.Functions.Worker
+namespace Microsoft.Azure.Functions.Worker.Http
 {
-    public class HttpResponseData
+    /// <summary>
+    /// A representation of the outgoing HTTP response.
+    /// </summary>
+    public abstract class HttpResponseData
     {
-        public HttpResponseData(HttpStatusCode statusCode, string? body = null)
-        {
-            StatusCode = statusCode;
-            Body = body;
-            Headers = new Dictionary<string, string>();
-        }
+        /// <summary>
+        /// Gets or sets the status code for the response.
+        /// </summary>
+        public abstract HttpStatusCode StatusCode { get; set; }
 
-        public HttpStatusCode StatusCode { get; set; }
+        /// <summary>
+        /// Gets or sets a <see cref="HttpHeadersCollection"/> containing the response headers
+        /// </summary>
+        public abstract HttpHeadersCollection Headers { get; set; }
 
-        // TODO: Custom body type (BodyContent)
-        public string? Body { get; set; }
+        /// <summary>
+        /// Gets or sets the response body stream.
+        /// </summary>
+        public abstract Stream Body { get; set; }
 
-        public Dictionary<string, string> Headers { get; set; }
+        /// <summary>
+        /// Gets an <see cref="HttpCookies"/> instance containing the request cookies.
+        /// </summary>
+        public abstract HttpCookies Cookies { get; }
```

Comment:
I think I remember talks about having some sort of static "CreateResponse" factory method attached here too in case folks tried to instantiate directly from HttpResponseData. Should that be included here?

Reply:
Yeah, I wanted to add that, so thanks for the reminder. Done.